src/sql/workbench/browser/modelComponents/inputbox.component.ts

```diff
@@ -141,7 +141,19 @@ export default class InputBoxComponent extends ComponentBase implements ICompone
 
 	public validate(): Thenable<boolean> {
 		return super.validate().then(valid => {
-			this.inputElement.validate();
+			valid = valid && this.inputElement.validate();
+
+			// set aria label based on validity of input
+			if (valid) {
+				this.inputElement.setAriaLabel(this.ariaLabel);
+			} else {
+				if (this.ariaLabel) {
+					this.inputElement.setAriaLabel(this.ariaLabel + nls.localize('period', ". ") + this.inputElement.inputElement.validationMessage);
```

Comment:
Suggest using format placeholders instead - it's cleaner and easier to read. 

```
this.inputElement.setAriaLabel(nls.localize('period', "{0}. {1}", this.ariaLabel, this.inputElement.inputElement.validationMessage));
```

Might also be useful to add a comment in this case since it's so generic - that can help the localizers understand the context of the period.

See https://github.com/microsoft/azuredatastudio/blob/87765e8673dcd00c71afbcc743af8d78c94a5b15/src/vs/workbench/contrib/terminal/common/terminalMenu.ts#L19 for an example.

Reply:
Definitely an improvement, will update now. Thanks @Charles-Gagnon!

Reply:
Better yet, template strings @Charles-Gagnon, @ellbosch

Reply:
Yes @Benjin!! This is great will update again.

Reply:
Per @Benjin's advice, changed to: ``this.inputElement.setAriaLabel(nls.localize('period', `${this.ariaLabel}. ${this.inputElement.inputElement.validationMessage}`));``

However I'm receiving the following warning from TS Lint: `Message argument to 'nls.localize' must be a string literal. (no-unexternalized-strings)tslint(1)`

How to proceed?

Reply:
Sadly localize requires you to pass in string literals - you can't use template strings for that